(chore) align React and Vue docs; align prop/param descriptions (#1814)
clerk/clerk-docs: docs/references/react/use-sign-in.mdx

## How to use the `useSignIn()` hook

### Check the current state of a sign-in

> Use `useSignIn()` to check the current sign-in status and display the value in the UI.

Reviewer: ahh actually, let's align it with the usesignup suggestion you added, to keep them consistent

```suggestion
Use `useSignIn()` to access the `SignIn` object and check the current state of a sign-up.
```

| `needs_second_factor` | The Second Factor verification is missing. The Second Factor is an optional step to provide additional verification and includes `phone_code` and `totp`. See [Second Factor](/docs/references/javascript/sign-in/second-factor) for details. |
| `needs_identifier` | The user's identifier (email address, phone number, username, etc.) hasn't been provided. |
| `needs_new_password` | The user needs to set a new password. |
<Properties>
  - `complete`
  - `string`

  The user has been signed in and the custom flow can proceed to `setActive()` to create a session.

  ---

  - `needs_first_factor`
  - `string`

  One of the following [first factor verification strategies](/docs/references/javascript/sign-in/first-factor) is missing: `email_link`, `email_code`, `phone_code`, `web3_metamask_signature`, `web3_coinbase_wallet_signature` or `oauth_provider`.

  ---

  - `needs_second_factor`
  - `string`

  One of the following [second factor verification strategies](/docs/references/javascript/sign-in/second-factor) is missing: `phone_code` or `totp`. This value is optional.

  ---

  - `needs_identifier`
  - `string`

> ### Create a custom sign-in flow with `useSignIn()`

Reviewer: these sections need to stay - they were added a result of user feedback.
we can rename to "Example" if that makes more sense

Author: https://github.com/clerk/clerk-docs/pull/1814/commits/70af8fedfe4a526a54f434837cf77c6e25921384
